src/app/components/chart/areaStackedTimeSeries.chart.directive.js

```diff
@@ -72,14 +79,23 @@
 
               d3.select('#' + scope.id + ' svg')
                 .datum(scope.data)
-                .style('height', height + 'px')
+                .style('height', scope.height + 'px')
                 .transition().duration(0)
                 .call(chart);
 
               return chart;
             });
 
             scope.$on('updateMetrics', function () {
+                if(scope.updatedID && scope.updatedID==scope.id){
```

Comment:
Also, since updatedID is never `nulled` or removed, I believe it will always try to redraw.